### PR TITLE
[ng] API homepage fixes; added angular2 pkg, added new filter categories, etc

### PR DESCRIPTION
**Pull request by chalin**
Solves all issues of #852 except for marking items as deprecated.

Staged at https://angulardart-org-dev.firebaseapp.com/angular/api

### CODE REVIEW

**Comment by kwalrath** [comment]
Looks great! We should add `[T] Typedef` to the "type" list, right?

**Comment by chalin** [comment]
Oh, I thought that was there! Thanks for catching that.

File: tools/api-builder/dart-package/services/preprocessDartDocData.js [hunk 1]
```diff
@@ -41,7 +41,8 @@ module.exports = function preprocessDartDocData(log, dartPkgConfigInfo) {
           const matches = e.href.match(/-([a-z]+)\.html/);
           let type = matches ? (e.typeFromHref = matches[1]) : e.type;
           // Conform to TS type names for now.
```
Comment (kwalrath) [inline tools/api-builder/dart-package/services/preprocessDartDocData.js]: Delete this comment?

**Comment by chalin** [comment]
> How hard would it be to implement redirects for things that used to be in angular2.common?

Depends at what granularity you'd like for it to be handled. Will you want redirects once we switch to 4.x too?  If so, do you want to open an issue for that?

**Comment by chalin** [comment]
Last commit just removed an obsolete comment. Travis was green before that. Merging.

**Comment by kwalrath** [comment]
>> How hard would it be to implement redirects for things that used to be in angular2.common?
>
> Depends at what granularity you'd like for it to be handled. Will you want redirects once we switch to 4.x too?  If so, do you want to open an issue for that?

Let me look into this a bit. I'm not sure it's worth the trouble.

File: tools/api-builder/dart-package/services/preprocessDartDocData.js (hunk 1 above)
Comment (chalin) [inline tools/api-builder/dart-package/services/preprocessDartDocData.js]: Yes, thanks. Done.

**Comment by kwalrath** [comment]
Looking at the analytics, the vast majority of API doc visits come in from Google searches, so once the Google index updates redirects would be irrelevant.

It'd be good to request an immediate re-indexing when we do change the URLs.

**Comment by chalin** [comment]
FYI, I've added a reminder to #670 concerning the re-indexing.

**Pull request merged**